project/ConstantHolderGenerator.scala

```diff
@@ -0,0 +1,44 @@
+import scala.annotation.tailrec
+
+import sbt._
+
+import org.scalajs.ir.ScalaJSVersions
+
+object ConstantHolderGenerator {
```

Comment:
We already use sbt-buildinfo for doing something similar, and I think one code generator thing is already too much :)